### PR TITLE
Release prep part 1

### PR DESCRIPTION
**Pull request by jim-parry**
- standardize "todo" tags in source
- remove unneeded autodoc.php
- remember composer update in admin/release
- first crack at upgrading 3.x to 4.x guide

### CODE REVIEW

File: user_guide_src/source/installation/upgrade_4xx.rst [hunk 1]
```diff
@@ -1,114 +1,102 @@
...
+- CI4 is built for PHP7.1+, and everything in the framework is namespaced, even the helpers
+
+**Application Structure**
+- The framework still has `application` and `system` folders, with the same 
+interpretation as before
+- The framework now provides for a `public` folder, intended as the document
+root for your app
+- There is also a `writable` folder, to hold cache data, logs, and session data
+- The `application` folder looks very similar to that for CI3, with some
+name changes (eg Filters instead of hooks) and capitalization, and some subfolders
```
Comment (lonnieezell) [inline user_guide_src/source/installation/upgrade_4xx.rst]: Events replaced Hooks, not filters.

File: user_guide_src/source/installation/upgrade_4xx.rst [hunk 2]
```diff
@@ -1,114 +1,102 @@
...
+See the threads in the `CodeIgniter 4 Roadmap <https://forum.codeigniter.com/forum-33.html>`_
+subforum for an uptodate list!
+
+**Downloads**
+- CI4 is still available as a ready-to-run zip or tarball, which
+includes the user guide (though in the `docs` subfolder
+- It can also be installed using Composer
+
+**Namespaces**
+- CI4 is built for PHP7.1+, and everything in the framework is namespaced, even the helpers
```
Comment (lonnieezell) [inline user_guide_src/source/installation/upgrade_4xx.rst]: Are helpers actually namespaced? We can do some loading of the files by parsing the namespace, but the file itself isn't actually namespaced, IIRC.

File: user_guide_src/source/installation/upgrade_4xx.rst [hunk 3]
```diff
@@ -1,114 +1,102 @@
...
+within the `App` (application) and `CodeIgniter` (i.e. system) top level
+namespaces; with composer autoloading support, and even using educated
+guessing to find your models and libraries if they are in the right
+folder even though not namespaced
+- You can configure the class loading to support whatever application structure
+you are most comfortable with, including the "HMVC" style
+
+**Controllers**
+- Controllers extend \CodeIgniter\Controller instead of CI_Controller
+- They don't use a constructor any more (to invoke CI "magic"); they
```
Comment (lonnieezell) [inline user_guide_src/source/installation/upgrade_4xx.rst]: I wouldn't mention `initController`. That is used behind the scenes to free up the constructor, but the user's don't need to know about it. The ones that know about it so far are just confused by it :)

File: user_guide_src/source/installation/upgrade_4xx.rst [hunk 4]
```diff
@@ -1,114 +1,102 @@
...
+automatic pagination
+- CI4 also has the `Entity` class you can build on, for
+richer data mapping to your database tables
+- Instead of CI3's `$this->load->model(x);`, you would now use
+`$this->x = new X();`, providing the fully-qualified path
+of your component
+
+**Views**
+- Your views look much like before, but they are invoked differently ...
+instead of CI3's `$this->load->view(x);` you can use `view(x);`
```
Comment (lonnieezell) [inline user_guide_src/source/installation/upgrade_4xx.rst]: You'll need to do `echo view()` not just `view()`

File: user_guide_src/source/installation/upgrade_4xx.rst (hunk 1 above)
Comment (jim-parry) [reply user_guide_src/source/installation/upgrade_4xx.rst]: I will stick to mentioning changed names :-/

File: user_guide_src/source/installation/upgrade_4xx.rst (hunk 2 above)
Comment (jim-parry) [reply user_guide_src/source/installation/upgrade_4xx.rst]: Oops - they are not.

File: user_guide_src/source/installation/upgrade_4xx.rst (hunk 3 above)
Comment (jim-parry) [reply user_guide_src/source/installation/upgrade_4xx.rst]: Will adjust wording

File: user_guide_src/source/installation/upgrade_4xx.rst (hunk 4 above)
Comment (jim-parry) [reply user_guide_src/source/installation/upgrade_4xx.rst]: Fixing :)

File: user_guide_src/source/installation/upgrade_4xx.rst [hunk 5]
```diff
@@ -55,8 +55,8 @@ you are most comfortable with, including the "HMVC" style
 
 **Controllers**
 - Controllers extend \CodeIgniter\Controller instead of CI_Controller
-- They don't use a constructor any more (to invoke CI "magic"); they
-instead call `initController`
+- They don't use a constructor any more (to invoke CI "magic") unless
+that is part of a bvase controller you make
```
Comment (lonnieezell) [inline user_guide_src/source/installation/upgrade_4xx.rst]: I'm guessing this was supposed to be `base` not `bvase` :)

**Comment by lonnieezell** [comment]
Looks good to me!

**Pull request merged**